pkg/stargate/deployments.go

```diff
@@ -347,7 +347,7 @@ func computeVolumes(template *api.StargateTemplate, dc *cassdcapi.CassandraDatac
 		VolumeSource: corev1.VolumeSource{
 			ConfigMap: &corev1.ConfigMapVolumeSource{
 				LocalObjectReference: corev1.LocalObjectReference{
-					Name: GeneratedConfigMapName(dc.ClusterName, dc.Name),
+					Name: GeneratedConfigMapName(dc.Spec.ClusterName, dc.Name),
```

Comment:
FYI, this fixes a subtle bug. `dc.ClusterName` refers to `dc.ObjectMeta.ClusterName` 😬